cpp/src/quantiles/quantiles_util.hpp

```diff
@@ -174,6 +174,71 @@ select_quantile(ValueAccessor get_value,
     }
 }
 
+template<typename Result, typename Iterator>
```

Comment:
Once the `exact` parameter is removed and that functionality is moved to a `quantiles_discrete`, function, I believe this type parameter can go away.